### PR TITLE
Add error logging for crc32 validation failures.

### PR DESCRIPTION
**Pull request by why-arong**
### Description

This PR add logging for situations where CRC32 validation does not pass.

By monitoring these failures, we can ensure data integrity throughout the application.

### CODE REVIEW

File: pymysqlreplication/event.py [hunk 1]
```diff
@@ -57,7 +58,14 @@ def _verify_event(self):
...
         byte_data = zlib.crc32(data).to_bytes(4, byteorder="little")
-        self._is_event_valid = True if byte_data == footer else False
+        if byte_data == footer:
+            self._is_event_valid = True
+        else:
+            self._is_event_valid = False
+            logging.error(
+                f"An CRC32 has failed for the event type {self.event_type}, "
+                "indicating a potential integrity issue with the data."
+            )
```
Comment (sean-k1) [inline pymysqlreplication/event.py:68]:
```suggestion
      self._is_event_valid = True if byte_data == footer else False
      if not self._is_event_valid:
            logging.error(
                f"An CRC32 has failed for the event type {self.event_type}, "
                "indicating a potential integrity issue with the data."
            )
```
@why-arong 
I think this looks more readable, what do you think?
I think the code you wrote is good, too.

Comment (why-arong) [reply pymysqlreplication/event.py:68]: I think that code reads better!

How about error message? Is there any additional information we can provide besides `'self.event_type'`??

**Comment by why-arong** [comment]
https://github.com/julien-duponchelle/python-mysql-replication/blob/1d2c8aba5d96ca4573054e51038d9f62b43da9e9/pymysqlreplication/binlogstream.py#L553-L558

How do you feel about not writing the code like `logging.log(logging.ERROR, ~~~)`? Unless I need to set the log level dynamically, I think `logging.error(~~~)` is more intuitive, so that's how I wrote it!

File: pymysqlreplication/event.py (hunk 1 above)
Comment (sean-k1) [reply pymysqlreplication/event.py:68]: I think that's enough

**Comment by sean-k1** [comment]
> @sean-k1 
> 
> https://github.com/julien-duponchelle/python-mysql-replication/blob/1d2c8aba5d96ca4573054e51038d9f62b43da9e9/pymysqlreplication/binlogstream.py#L553-L558
> 
> 
> 
> How do you feel about not writing the code like `logging.log(logging.ERROR, ~~~)`? Unless I need to set the log level dynamically, I think `logging.error(~~~)` is more intuitive, so that's how I wrote it!

logging.error is more better👍

**Review by sean-k1 — APPROVED** [review]
LGTM Thanks your hard work!

**Pull request merged**